spec/acceptance/mod_passenger_spec.rb

```diff
@@ -86,123 +171,175 @@
   EOS
 
   case fact('osfamily')
-  when 'Debian'
-    context "default passenger config" do
-      it 'succeeds in puppeting passenger' do
-        pp = <<-EOS
+    when 'Debian'
+      context "setting passenger options within the apache 'Directory' directive" do
+       it 'should allow something with no error' do
+         all_passenger_directory_options = passenger_config_options.select {|k,v| /directory/ =~ v[:Context]}
+         passenger_directory_options = ''
+         all_passenger_directory_options.each do |k,v|
+           passenger_directory_options << "'%s' => '%s',\n" % [k,'something']
+         end
+          pp = <<-EOS
+          class { 'apache': service_ensure => stopped }
+          class { 'apache::mod::passenger': }
+          /* a simple ruby rack 'hello world' app */
+          file { '/var/www/passenger':
+           ensure => directory,
+           owner  => '#{rackapp_user}',
+           group  => '#{rackapp_group}',
+          }
+          file { '/var/www/passenger/config.ru':
+            ensure  => file,
+            owner   => '#{rackapp_user}',
+            group   => '#{rackapp_group}',
+            content => "app = proc { |env| [200, { \\"Content-Type\\" => \\"text/html\\" }, [\\"hello <b>world</b>\\"]] }\\nrun app",
+          }
+          apache::vhost { 'passenger.example.com':
+            port          => '80',
+            docroot       => '/var/www/passenger/public',
+            docroot_group => '#{rackapp_group}',
+            docroot_owner => '#{rackapp_user}',
+            directories => [
+               { 'path' => '/var/www/passenger',
+                 #{passenger_directory_options}
+               }     
+            ],
+            require       => File['/var/www/passenger/config.ru'],
+          }
+          host { 'passenger.example.com': ip => '127.0.0.1', }
+          EOS
+         apply_manifest(pp, :catch_failures => true)
+       end
+       describe file("#{$vhost_dir}/25-passenger.example.com.conf") do
+         all_passenger_directory_options = passenger_config_options.select {|k,v| /directory/ =~ v[:Context]}
+         all_passenger_directory_options.each do |k,v|
+           case v[:Type]
+             when 'QuotedString', 'RelPath', 'FullPath', 'Path'
+              it { is_expected.to contain "#{v[:PassOpt]} \"something\"" }
+             when 'String', 'URI', 'Integer'
```

Comment:
It's not clear to me why `String`, `URI`,  and `Integer` need their own case? It looks like the code is functionally the same to the else statement below.